spring-integration-mongodb/src/main/java/org/springframework/integration/mongodb/outbound/MongoDbOutboundGateway.java

```diff
@@ -0,0 +1,171 @@
+import org.springframework.integration.handler.AbstractReplyProducingMessageHandler;
+import org.springframework.messaging.Message;
+import org.springframework.util.Assert;
+
+/**
+ * Makes outbound operations to query a MongoDb database using a {@link MongoOperations}
+ *
+ * @author Xavier Padró
+ * @since 5.0
+ */
+public class MongoDbOutboundGateway extends AbstractReplyProducingMessageHandler {
+
+	private volatile EvaluationContext evaluationContext;
+
+	private volatile Expression queryExpression;
+
+	private volatile MongoOperations mongoTemplate;
+
+	private volatile MongoDbFactory mongoDbFactory;
+
+	private volatile MongoConverter mongoConverter;
+
+	private volatile boolean expectSingleResult = false;
+
+	private volatile Class<?> entityClass = Document.class;
+
+	private volatile Expression collectionNameExpression;
+
+	public MongoDbOutboundGateway(MongoDbFactory mongoDbFactory) {
+		this(mongoDbFactory, new MappingMongoConverter(new DefaultDbRefResolver(mongoDbFactory),
+				new MongoMappingContext()));
+	}
+
+	public MongoDbOutboundGateway(MongoDbFactory mongoDbFactory, MongoConverter mongoConverter) {
+		Assert.notNull(mongoDbFactory, "mongoDbFactory must not be null.");
+		Assert.notNull(mongoConverter, "mongoConverter must not be null.");
+		this.mongoDbFactory = mongoDbFactory;
+		this.mongoConverter = mongoConverter;
+	}
+
+	public MongoDbOutboundGateway(MongoOperations mongoTemplate) {
+		Assert.notNull(mongoTemplate, "mongoTemplate must not be null.");
+		this.mongoTemplate = mongoTemplate;
+	}
+
+	public void setQueryExpression(Expression queryExpression) {
+		Assert.notNull(queryExpression, "queryExpression must not be null.");
+		this.queryExpression = queryExpression;
+	}
+
+	public void setExpectSingleResult(boolean expectSingleResult) {
+		this.expectSingleResult = expectSingleResult;
+	}
+
+	public void setEntityClass(Class<?> entityClass) {
+		Assert.notNull(entityClass, "entityClass must not be null.");
+		this.entityClass = entityClass;
+	}
+
+	public void setCollectionNameExpression(Expression collectionNameExpression) {
+		Assert.notNull(collectionNameExpression, "collectionNameExpression must not be null.");
+		this.collectionNameExpression = collectionNameExpression;
+	}
+
+	public void setCollectionNameExpressionString(String collectionNameExpressionString) {
+		Assert.notNull(collectionNameExpressionString, "collectionNameExpressionString must not be null.");
+		this.collectionNameExpression = EXPRESSION_PARSER.parseExpression(collectionNameExpressionString);
+	}
+
+	public void setMongoConverter(MongoConverter mongoConverter) {
+		Assert.notNull(mongoConverter, "mongoConverter cannot be null");
+		Assert.isNull(this.mongoTemplate,
+				"'mongoConverter' can not be set when instance was constructed with MongoTemplate");
+		this.mongoConverter = mongoConverter;
+	}
+
+	@Override
+	protected void doInit() {
+		if (this.queryExpression == null) {
+			throw new IllegalStateException("no query specified");
```

Comment:
@xpadro ,

Thank you for so prompt input! 
Well, looks like Travis is dead. Not our business. At least for now. 😄 

M-m-m.
No, I meant `Assert.state()`. Why didn't it work for you ?